### PR TITLE
Refactor Validation Logic to validators, unit testing

### PR DESCRIPTION
**Pull request by ColeDewis**
Updates validation logic for events + attendees to use a separate object, which is then also unit tested.

I've added Ethan and Carlin to review, since they wrote the validation and so may see issues with my changes, or may know of any places I've missed

### CODE REVIEW

**Review by ethanchung936 — COMMENTED** [review]
When the fields aren't filled in, the error message is for the last field that isn't filled, I initially intended it to be the first one not filled out so it went in order. The validation still works though so it seems fine. If we do want to change it, could put the conditions in the opposite order or return as soon as an empty field is encountered

**Comment by ColeDewis** [comment]
> When the fields aren't filled in, the error message is for the last field that isn't filled, I initially intended it to be the first one not filled out so it went in order. The validation still works though so it seems fine. If we do want to change it, could put the conditions in the opposite order or return as soon as an empty field is encountered

That's a fair point - I'll edit the logic to do that

File: code/noRAM/app/src/main/java/com/example/noram/EventValidator.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+package com.example.noram;
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/EventValidator.java:4]: file comment

File: code/noRAM/app/src/main/java/com/example/noram/AttendeeValidator.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package com.example.noram;
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/AttendeeValidator.java:4]: file comment

File: code/noRAM/app/src/test/java/com/example/noram/EventUnitTest.java [hunk 3]
```diff
@@ -1,18 +1,18 @@
 package com.example.noram;
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/test/java/com/example/noram/EventUnitTest.java:2]: comment

**Review by WhoShotCupidd — APPROVED** [review]
lgtm

**Pull request merged**